2024/MIlano-Pacini

```diff
@@ -0,0 +1,98 @@
+import re
+import requests
+from datetime import datetime, timedelta
+import pytz
+
+
+# Espressione regolare per estrarre l'IP e il timestamp
+log_pattern = r"(?P<ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}) - - \[(?P<timestamp>\d{2}/[A-Za-z]{3}/\d{4}:\d{2}:\d{2}:\d{2} [+\-]\d{4})\] \"(?P<request>[^\"]+)\" (?P<status>\d+)"
+
+# Fuso orario italiano
+italian_tz = pytz.timezone("Europe/Rome")
+
+# Variabili per memorizzare i gruppi
+gruppi = {}
+inizio_primo_gruppo = None
+gruppo_corrente = 1
+
+
+# get per verificare se l'ip è situtato in italia
+def is_ip_from_italy(ip):
+    try:
+        response = requests.get(f"https://ipinfo.io/{ip}/json")
+        data = response.json()
+        return data.get("country") == "IT"
+    except Exception as e:
+        print(f"Errore durante la geolocalizzazione dell'IP {ip}: {e}")
+        return False
+
+
+# lettura file e organizzazione in gruppi
+with open("www.ntop.org-access.log.13", "r") as file:
```

Comment:
Dovete leggere i files di log da una cartella in ordine cronologico/di nome e NON va mai cablato niente nel codice